python/cudf/cudf/io/parquet.py

```diff
@@ -680,7 +702,29 @@ def to_parquet(
     """{docstring}"""
 
     if engine == "cudf":
+        # Ensure that no columns dtype is 'category'
+        for col in df.columns:
+            if df[col].dtype.name == "category":
+                raise ValueError(
+                    "'category' column dtypes are currently not "
+                    + "supported by the gpu accelerated parquet writer"
+                )
+
         if partition_cols:
+            if metadata_file_path is not None:
+                warnings.warn(
+                    "metadata_file_path will be ignored/overwritten when "
+                    "partition_cols are provided"
+                )
```

Comment:
Can you explain this? It seems like there may be a significant change in behavior here. If so, the behavior should probably be described in the docstring.

Reply:
Ah - I see. This is a consequence of the ugly circular logic between `to_parquet` and `write_to_dataset`. Some historical context: I believe `write_to_dataset` was originally inteded to be the only way to write out a partitioned dataset. However, `partition_cols` was later added to `to_parquet` to redirect back into `write_to_dataset`.  The problem with this latter approach is that you can no longer specify `metadata_file_path`. Instead, the user must specify `return_metadata=True` if they want the final metadata object to be returned.

Perhaps we need to add the `return_metadata` kwarg to the `to_parquet` docstring to make this clearer? Either way, the warning above should probably suggest using `return_metadata=True` if the user tries specifying `metadata_file_path`. What do you think?

Reply:
The problem created by this PR is that now there are two ways to get partitioning with `to_parquet`. The first is to  pass `partition_cols` and `partition_file_name` and let cudf figure out the partitions and directory structure. The other is to pass a list of paths, offsets into the df to partition by and list of metadata_file_paths to set to each partition. 

The second method is not really intended to be used by users and I worry that someone might think to pass args for both and get confused.